x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,495 @@
+
+    let actor = context.actor();
+    let voters: Vec<_> = (0..proposal.voters_len)
+        .map(|i| {
+            context
+                .get(StateKeys::Voter(id, i as usize))
+                .unwrap()
+                .unwrap()
+        })
+        .collect();
+
+    let i = _ensure_voter(&actor, &voters)?;
+
+    if yea {
+        context
+            .store_by_key(
+                StateKeys::Yeas(id),
+                &(context.get::<u64>(StateKeys::Yeas(id)).unwrap().unwrap() + 1),
+            )
+            .unwrap();
+    } else {
+        context
+            .store_by_key(
+                StateKeys::Nays(id),
+                &(context.get::<u64>(StateKeys::Nays(id)).unwrap().unwrap() + 1),
+            )
+            .unwrap();
+    }
+
+    context
+        .store_by_key(
+            StateKeys::Voter(id, i as usize),
+            &Voter {
+                address: actor,
+                voted: true,
+            },
+        )
+        .unwrap();
+
+    Ok(())
+}
+
+#[public]
+pub fn execute(
+    context: Context<StateKeys>,
+    proposal_id: u64,
+    max_units: u64,
+) -> Result<DeferDeserialize, ProposalError> {
+    let mut proposal =
+        _proposal_at(&context, proposal_id).ok_or(ProposalError::InexistentProposal)?;
+
+    if proposal.executed {
+        return Err(ProposalError::AlreadyExecuted);
+    }
+
+    let yeas = context.get(StateKeys::Yeas(proposal_id)).unwrap().unwrap();
+    let nays = context.get(StateKeys::Nays(proposal_id)).unwrap().unwrap();
+    if !_quorum_reached(proposal.voters_len, yeas, nays) {
+        return Err(ProposalError::QuorumNotReached);
+    }
+
+    proposal.executed = true;
+
+    proposal
+        .to
+        .call_function::<DeferDeserialize>(
+            &proposal.method,
+            &proposal.args,
+            max_units,
+            proposal.value,
+        )
+        .map_err(ProposalError::ExecutionFailed)
+}
+
+#[public]
+pub fn last_proposal_id(context: Context<StateKeys>) -> u64 {
+    _last_proposal_id(&context)
+}
+
+#[public]
+pub fn quorum_reached(
+    context: Context<StateKeys>,
+    proposal_id: u64,
+) -> Result<bool, ProposalError> {
+    let proposal = _proposal_at(&context, proposal_id).ok_or(ProposalError::InexistentProposal)?;
+    let yeas = context.get(StateKeys::Yeas(proposal_id)).unwrap().unwrap();
+    let nays = context.get(StateKeys::Nays(proposal_id)).unwrap().unwrap();
+
+    Ok(_quorum_reached(proposal.voters_len, yeas, nays))
+}
+
+fn _proposal_at(context: &Context<StateKeys>, proposal_id: u64) -> Option<Proposal> {
+    context
+        .get(StateKeys::Proposals(proposal_id))
+        .expect("state corrupt")
+}
+
+fn _last_proposal_id(context: &Context<StateKeys>) -> u64 {
+    context
+        .get(StateKeys::LastProposalId)
+        .expect("state corrupt")
+        .unwrap_or_default()
+}
+
+fn _quorum_reached(max_votes: u64, yeas: u64, nays: u64) -> bool {
+    yeas + nays >= MIN_VOTES && yeas > nays && yeas >= (max_votes + 1) / 2
+}
```

Comment:
Please put all of these (and any others) in a `mod internal` (same file) and remove the prefixes.

Reply:
Thanks, namespaced these functions in the `internal` module.